File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        l = 0
        r = 0
        while l < len(nums):
            if nums[l] != 0:
                nums[r], nums[l] = nums[l], nums[r]
                r += 1
                l += 1
            else:
                l += 1
            

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int l = 0;
        int r = 0;
        while(l < nums.size()){
            if(nums[l] != 0){
                int temp = nums[r];
                nums[r] = nums[l];
                nums[l] = temp;
                r++;
                l++;
            }
            else{
                l++;
            }
        }
    }
};